src/choose-location/choose-location.components.test.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { of } from "rxjs";
+import { mount } from "enzyme";
+import ChooseLocation from "./choose-location.component";
+import {
+  getLoginLocations,
+  setSessionLocation
+} from "./choose-location.resource";
+import { act } from "react-dom/test-utils";
+
+const historyMock = { push: jest.fn() };
+const mockedGetLoginLocations = getLoginLocations as jest.Mock;
+const mockedSetSessionLocation = setSessionLocation as jest.Mock;
+
+const mockedChooseLocationResources = {
+  getLoginLocations: () =>
+    of({ uuid: "111", display: "Earth" }, { uuid: "222", display: "Mars" }),
+  setSessionLocation: () => Promise.resolve()
+};
+
+jest.doMock("./choose-location.resource", () => mockedChooseLocationResources);
+
+//jest.mock("./choose-location.resource", () => ({
+//getLoginLocations: jest.fn(),
+//setSessionLocation: jest.fn()
+//}));
```

Comment:
I couldn't figure out how to mock getLoginLocations.

Reply:
Spent another couple hours wrestling with it. Split the component into presentational and container components to make injection easier, until I reviewed [Dan's article](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0) and saw the new note at the top. Time to go read documentation for a while.